src/extractor/extractor.cpp

```diff
@@ -114,8 +115,43 @@ int Extractor::run(ScriptingEnvironment &scripting_environment)
     const unsigned recommended_num_threads = tbb::task_scheduler_init::default_num_threads();
     const auto number_of_threads = std::min(recommended_num_threads, config.requested_num_threads);
     tbb::task_scheduler_init init(number_of_threads);
-
     {
+        // Node Location Cache initialize
+        std::unique_ptr<index_sparse_type> index_sparse;
+        std::unique_ptr<index_dense_type> index_dense;
+        std::unique_ptr<location_handler_sparse_type> location_handler_sparse;
+        std::unique_ptr<location_handler_dense_type> location_handler_dense;
+
+        std::string cacheName(config.cache_file_path);
+        cacheName += "nodes.cache";
+        int fd = open(cacheName.c_str(), O_RDWR | O_CREAT, 0666);
+        if (fd == -1)
+        {
+            throw std::runtime_error(strerror(errno));
+        }
+
+        if("NONE" != config.cache_file_type)
+        {
+
+
+            if("SPARSE" == config.cache_file_type)
+            {
+                std::cout << "SPARSE" << std::endl;
+                index_sparse.reset(new index_sparse_type{fd});
+                location_handler_sparse.reset(new location_handler_sparse_type(*index_sparse.get()));
+                location_handler_sparse->ignore_errors();
+                scripting_environment.setSparseCache(*location_handler_sparse);
+            }
+            else
+            {
+                std::cout << "DENSE" << std::endl;
+                index_dense.reset(new index_dense_type{fd});
+                location_handler_dense.reset(new location_handler_dense_type(*index_dense.get()));
+                location_handler_dense->ignore_errors();
+                scripting_environment.setDenseCache(*location_handler_dense);
+            }
+        }
```

Comment:
The indices should be default constructible, so there is no need for the unique ptr. Also it would be great to not having to parse strings here. These should be typed options set in the config.

    Index ind;

    if (config.use_sparse)
      ind = makeIndex();